From 0da7f8cabcb7510ef890e4f97ca7c5f8458e5f6c Mon Sep 17 00:00:00 2001
From: Vinith Kumar R <quic_vinithku@quicinc.com>
Date: Tue, 1 Oct 2024 12:14:42 +0530
Subject: [PATCH 2/4] wifi: ath12k: Add NULL check for key before accessing
 inside the ieee80211_xmit_fast_finish

Currently, in the ieee80211_xmit_fast_finish function, variable key
is dereferred. This will cause a crash if key is NULL. So avoid
the possible exception case by adding a NULL check
for key before the usage

Signed-off-by: Vinith Kumar R <quic_vinithku@quicinc.com>
---
 net/mac80211/tx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/tx.c b/net/mac80211/tx.c
index cbe0568..0d0d527 100644
--- a/net/mac80211/tx.c
+++ b/net/mac80211/tx.c
@@ -3653,7 +3653,7 @@ ieee80211_xmit_fast_finish(struct ieee80211_sub_if_data *sdata,
 	sta->deflink.tx_stats.bytes[skb_get_queue_mapping(skb)] += skb->len;
 	sta->deflink.tx_stats.packets[skb_get_queue_mapping(skb)]++;
 
-	if (pn_offs) {
+	if (pn_offs && key) {
 		u64 pn;
 		u8 *crypto_hdr = skb->data + pn_offs;
 
-- 
2.34.1

